src/training/data_preprocessing.py

```diff
@@ -0,0 +1,105 @@
+"""Data preprocessing for original rating matrix."""
+
+import numpy as np
+import json
+import os
+from scipy import sparse
+from src.data_store.local_data_store import LocalDataStore
+from src.config import (HPF_SCORING_REGION,
+                        HPF_input_raw_data)
+from src.utils import (cal_sparsity)
+import logging
+
+logging.basicConfig()
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+class DataPreprocessing:
+    """Data preprocessing for original rating matrix."""
+
+    def __init__(self, datastore=None,
+                 scoring_region=HPF_SCORING_REGION):
+        """Intialize preprocessing variables."""
+        self.datastore = datastore
+        self.scoring_region = scoring_region
+        self.trimmed_manifest_list = None
+        self.package_id_dict = {}
+        self.manifest_id_dict = {}
+        self.rating_matrix = None
+        self.loadS3()
+
+    def loadS3(self):
+        """Load the raw manifest.json from S3."""
+        manifest_json_filename = os.path.join(
+            self.scoring_region, HPF_input_raw_data)
+        all_manifest_list = self.datastore.read_json_file(
+            manifest_json_filename)
+        all_manifest_list = all_manifest_list[0].get('package_list', [])
+        logger.info("Number of manifests collected = {}".format(
+            len(all_manifest_list)))
+        self.trimmed_manifest_list = [
+            manifest for manifest in all_manifest_list if 13 < len(manifest) < 15]
+        logger.info("Number of trimmed manifest = {}". format(
+            len(self.trimmed_manifest_list)))
+        del(all_manifest_list)
+
+    def generate_package_id_dict(self):
```

Comment:
I would much rather have you use this operation:

```python
dict(zip(list(range(len(trimmed_package_list))), trimmed_package_list))
```
Don't do it all in one line like I did though 🙂